chatbot-core/embedding.py

```diff
@@ -0,0 +1,112 @@
+from dataclasses import dataclass, field
+from typing import List
+
+import numpy as np
+from langchain_community.embeddings import OllamaEmbeddings
+from langchain_core.embeddings import Embeddings
+from langchain_openai import OpenAIEmbeddings
+from sentence_transformers import SentenceTransformer
+from torch import Tensor
+
+# Note to check if the vector dimensions are same (impt for injecting in vector db)
+
+
+@dataclass
+class SentenceTransformerEmbeddingModel(Embeddings):
+    """
+    Embedding model using Sentence Transformers
+    """
+
+    model: str = "sentence-transformers/paraphrase-multilingual-MiniLM-L12-v2"
+    embedding_model: SentenceTransformer = field(init=False)
+
+    def __post_init__(self):
+        """
+        Post initialization
+        """
+        self.st_embedding_model = SentenceTransformer(self.model)
+
+    def embed_documents(self, texts: list) -> np.ndarray:
+        """
+        Generate embeddings for a list of documents
+        """
+        return self.st_embedding_model.encode(texts)
+
+    def embed_query(self, text: str) -> np.ndarray:
+        """
+        Generate embedding for a piece of text
+        """
+        return self.st_embedding_model.encode(text)
+
+    def check_similarity(self, embeddings_1: np.ndarray, embeddings_2: np.ndarray) -> Tensor:
+        """
+        Computes the cosine similarity between two embeddings
+        """
+        return self.st_embedding_model.similarity(embeddings_1, embeddings_2)
+
+    def get_model(self):
+        """Returns the model"""
+        return self.st_embedding_model
+
+
+@dataclass
+class OllamaEmbeddingModel(Embeddings):
+    """
+    Embedding model using Ollama (locally deployed)
+    """
+
+    model: str = "all-minilm:latest"
+    base_url: str = "http://localhost:11434"
```

Comment:
Let's not use default values.

```suggestion
    base_url: str
```